src/rgw/rgw_rados.cc

```diff
@@ -5586,7 +5586,7 @@ int RGWRados::Bucket::List::list_objects(int64_t max,
         continue;
       }
       bool check_ns = (obj.ns == params.ns);
-      if (!params.list_versions && !entry.is_visible()) {
+      if ((!params.list_versions && !entry.is_visible()) || (!params.list_versions && (params.marker == index_key.name))) {
```

Comment:
`rgw_bucket_list()` in `cls/rgw/cls_rgw.cc` has this same logic:
```c++
      if (!op.list_versions && !entry.is_visible()) {
```
because cls is already doing this filtering for us, I would agree with @mattbenjamin that `cls_rgw.cc` is the right place to do this extra filtering. could you please try it that way?

Reply:
hi , @cbodley ,i  updated the code   :）